database/engine/global_uuid_map/global_uuid_map.c

```diff
@@ -0,0 +1,76 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+
+#include "global_uuid_map.h"
+
+static Pvoid_t JGUID_map = (Pvoid_t) NULL;
+static Pvoid_t JGUID_object_map = (Pvoid_t) NULL;
+
+int guid_store(uuid_t uuid, char *object)
+{
+#ifdef NETDATA_INTERNAL_CHECKS
+    static uint32_t count = 0;
+#endif
+    if (unlikely(!object))
+        return 0;
+
+    Pvoid_t *PValue;
+
+    PValue = JudyHSIns(&JGUID_map, (void *) uuid, (Word_t) sizeof(uuid_t), PJE0);
+    if (*PValue)
```

Comment:
This will crash if Judy returns an error condition. For example if `uuid==0` then it will trigger an internal error inside `JudyLIns` which returns `~0`. Always dereferencing seems dangerous - should this be `if (PValue!=PPJERR && *PValue)` ?

Reply:
Currently netdata always crashes when out of memory like in:
https://github.com/netdata/netdata/blob/98c7260d929a2f6ad2aabb4b27f0d77d49f6cae8/libnetdata/libnetdata.c#L86

We are consistently crashing in dbengine as well, but we're not logging it similarly.
If we are to be consistent we could add:
```
    if (PPJERR==PValue) fatal("JudyHSIns() cannot allocate memory.");
```

Reply:
This would cover the out of memory case, but there are multiple conditions in Judy that trigger that error return. The other one that I saw in the review was parameter error. In this case we could guard against the parameter error explicitly, or guard against the error being returned from Judy. I think guarding against the error returning from Judy is overall more robust.

Reply:
The `if (PPJERR==PValue) fatal ...` has been added.